WeatherDataCalculator.py

```diff
@@ -0,0 +1,82 @@
+
+
+class WeatherDataCalculator:
+    def __init__(self, weather_data_calculations, weather_data):
+        self.weather_data_calculations = weather_data_calculations
+        self.weather_data = weather_data
+
+    def calculate_year_results(self, year):
+        if year in self.weather_data.keys():
+            max_temp, max_month, max_day, min_temp, min_month, min_day, max_humidity, max_humidity_month, max_humidity_day = \
+                self.calculate_temperatures(year)
+            self.weather_data_calculations[year] = {
+                "max_temp": max_temp,
+                "max_month": max_month,
+                "max_day": max_day,
+                "min_temp": min_temp,
+                "min_month": min_month,
+                "min_day": min_day,
+                "max_humidity": max_humidity,
+                "max_humidity_month": max_humidity_month,
+                "max_humidity_day": max_humidity_day
+            }
```

Comment:
Only save required fields at the time of file reading and save a single entry in dict